app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -66,12 +67,29 @@ public String getInitialUrl() {
         return getArguments().getString(ARGUMENT_URL);
     }
 
+    private void updateURL(final String urlString) {
+        try {
+            URI uri = new URI(urlString);
+            final String userInfo = uri.getUserInfo();
+            if (userInfo != null) {
+                // Strip the userInfo to minimise spoofing ability. This only affects what's shown
+                // during browsing, this information isn't used when we start editing the URL:
+                uri = new URI(uri.getScheme(), null, uri.getHost(), uri.getPort(), uri.getPath(), uri.getQuery(), uri.getFragment());
+            }
+
+            urlView.setText(uri.toString());
+        } catch (URISyntaxException e) {
+            // WebView should always have supplied a valid URL
+            throw new IllegalStateException("WebView is expected to always supply a valid URL");
```

Comment:
I wonder if we should only crash on beta/debug and return the original URL on release.